src/lib/consensus/proof_of_stake.ml

```diff
@@ -2062,9 +2062,15 @@ module Data = struct
              ~f:(fun diff -> Ok diff)
       in
       let%map total_currency =
-        Amount.add previous_consensus_state.total_currency supply_increase
-        |> Option.map ~f:Or_error.return
-        |> Option.value
+        Option.Let_syntax.(
```

Comment:
Nit: `let open Option.Let_syntax in` might cause less aggressive indentation here.

Reply:
Yeah I'll rename it. TBH medium term I want to refactor this a bit since it's slightly incoherent at the moment. Specifically, I think the "ledger" type should really contain all the data necessary to apply a transaction.